src/sonic-config-engine/minigraph.py

```diff
@@ -873,8 +942,17 @@ def parse_device_desc_xml(filename):
 
     return results
 
+def parse_asic_sub_role(filename, asic_name):
+    if not os.path.isfile(filename):
+        return None
+    root = ET.parse(filename).getroot()
+    for child in root:
+        if child.tag == str(QName(ns, "MetadataDeclaration")):
+             _, _, _, _, _, _, _, sub_role = parse_meta(child, asic_name)
```

Comment:
since asic meta is quite different from device meta, I suggest not to reuse parse_meta. introduce a new function call parse_asic_meta

Reply:
Added new function.